pkg/resolution/resolver/git/resolver.go

```diff
@@ -157,7 +159,7 @@ func validateRepoURL(url string) bool {
 	return re.MatchString(url)
 }
 
-func ResolveAnonymousGit(ctx context.Context, params map[string]string) (framework.ResolvedResource, error) {
+func ResolveAnonymousGit(ctx context.Context, params map[string]string, kubeclient kubernetes.Interface, logger *zap.SugaredLogger, cache *cache.LRUExpireCache, ttl time.Duration) (framework.ResolvedResource, error) {
```

Comment:
just a thought, would make sense to pass the kubeclient, cache and ttl through a single struct. A lot of arguments are being passed to a single function otherwise.

Reply:
Yeah, this is a number of additional parameters which are only necessary if there is a git token specified. Would it make more sense to instead require `ResolveGit` instead be given a `plumbing.transport.AuthMethod`? this way, the git resolver method is ignorant about kubernetes, secrets caching, etc, and the caller function is responsible for determining if `plumbTransport.AuthMethod(nil)` is correct, parsing the secret, and constructing the `http.BasicAuth` if necessary?

Reply:
it's annoying tbh because i need to refactor the whole code which could have been in another PR by someone else... but anyway i'll do it this time..

Reply:
@chmouel thank you for the help